google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/ManagedCloudSdkUpdater.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.sdk;
+
+import com.google.cloud.tools.intellij.GctFeature;
+import com.google.cloud.tools.intellij.service.PluginInfoService;
+import com.google.common.annotations.VisibleForTesting;
+import com.intellij.notification.Notification;
+import com.intellij.openapi.components.ServiceManager;
+import java.awt.event.ActionListener;
+import java.time.Clock;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/** Schedules and manages automatic updates for {@link ManagedCloudSdkService}. */
+public class ManagedCloudSdkUpdater {
```

Comment:
would the be better as an application service? (and if so update the name accordingly - ManagedCloudSdkUpdateService or something like that)

Reply:
I kind of see application service as something useful for at least some parts of the whole application. This is a pure helper class for Managed SDK, so I wasn't sure about that, adding another service just for testing purposes.. What do you think, does it feel much cleaner if we put such classes into plugin config?

Reply:
I don't have a very strong opinion on this, plus it seems the rest of the code is relying a lot on service manager, I will update it as a service so it will be in line.